graylog2-server/src/main/java/org/graylog/plugins/views/search/jobs/periodical/SearchJobStateCleanupPeriodical.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.views.search.jobs.periodical;
+
+import jakarta.inject.Inject;
+import org.graylog.plugins.views.search.jobs.SearchJobStateService;
+import org.graylog2.plugin.periodical.Periodical;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SearchJobStateCleanupPeriodical extends Periodical {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SearchJobStateCleanupPeriodical.class);
+    static final Duration MIN_AGE_TO_REMOVE = Duration.standardDays(2);
+    static final Duration MIN_AGE_TO_EXPIRE = Duration.standardDays(1);
+
+    private final SearchJobStateService searchJobStateService;
+
+    @Inject
+    public SearchJobStateCleanupPeriodical(final SearchJobStateService searchJobStateService) {
+        this.searchJobStateService = searchJobStateService;
+    }
+
+    @Override
+    public void doRun() {
+        removeOldJobs();
+        expireOldEnoughJobs();
+    }
+
+    private void removeOldJobs() {
+        final DateTime dateTime = DateTime.now(DateTimeZone.UTC).minus(MIN_AGE_TO_REMOVE);
```

Comment:
Would like to clear this with product before merging it

Reply:
I'd say that during development/experimentation it is not problematic. We can switch easily to different approach after discussion with product, and it won't cause any db/migration problems.